Jdp190401 42

KucharM opened the pull request:
implementacja metod w klasach service

KucharM committed:
implementacja metod w Dao

KucharM committed:
implementacja klas service

klikuskodilla commented on src/main/java/com/kodilla/ecommercee/dao/CartDao.java:
```diff
@@ -3,9 +3,12 @@
 
 @Repository
-@Transactional
 public interface CartDao extends CrudRepository<Cart, Long> {
```
findAll ma sens, reszta metod jest dokładnie taka sama jak implementacja z CrudRepository

klikuskodilla commented on src/main/java/com/kodilla/ecommercee/dao/ProductDao.java:
```diff
@@ -3,10 +3,13 @@
 
 @Repository
-@Transactional
 public interface ProductDao extends CrudRepository< Product, Long > {
```
uwagi idenetyczne jak w CartDao

klikuskodilla commented on src/main/java/com/kodilla/ecommercee/dao/ProductGroupDao.java:
```diff
@@ -3,9 +3,13 @@
 
 @Repository
-@Transactional
 public interface ProductGroupDao extends CrudRepository<ProductGroup, Long> {
```
uwagi idenetyczne jak w CartDao

klikuskodilla commented on src/main/java/com/kodilla/ecommercee/dao/UserDao.java:
```diff
@@ -3,9 +3,14 @@
 
 @Repository
-@Transactional
 public interface UserDao extends CrudRepository<User, Long> {
```
uwagi idenetyczne jak w CartDao

klikuskodilla commented on src/main/java/com/kodilla/ecommercee/service/OrderService.java:
```diff
@@ -1,4 +1,32 @@
+import java.util.Optional;
+
+@Service
+@Transactional
```
Tutaj nie powinien zostać użyty mapper?

KucharM replied:
myślałem że będziemy mapować dopiero w kontrolerach, a w klasach service zrobimy tylko takie opakowanie dla metod wyciągniętych z dao,

klikuskodilla commented on src/main/java/com/kodilla/ecommercee/service/ProductGroupService.java:
```diff
@@ -1,4 +1,28 @@
+import java.util.Optional;
+
+@Service
+@Transactional
```
Po co Transactional

klikuskodilla commented on src/main/java/com/kodilla/ecommercee/service/ProductGroupService.java:
```diff
@@ -1,4 +1,28 @@
+@Service
+@Transactional
 public class ProductGroupService {
+    @Autowired
```
Tutaj trzeba mapowa z/do Dto

Makedo81 replied:
Nie wiem czy ten task jest tak naprawde jeszcze potrzebny, bo jestesmy juz na etapie implementacji kontrolerow i powiazanych z nimi maperr i service,wiec po czesci jest to juz zrobione na indywidualnych branch

klikuskodilla reviewed (changes requested):
Po co anotacja Transactional i dlaczego nie mapujecie z/do DTO?

klikuskodilla commented:
Dlaczego tutaj znajdują się commity z innego taska?

KucharM committed:
Merge branch 'master' into JDP190401-42

KucharM committed:
implementacja metod dao

klikuskodilla reviewed (changes requested):
Nie dostałem odpowiedzi na cześć uwag

klikuskodilla committed:
Merge branch 'master' into JDP190401-42

klikuskodilla committed:
Merge branch 'master' into JDP190401-42

pszczepcio committed:
1). OrderDaoTestSuite:
- usunięcie cartDao.deletById() (ta encja jest usuwana wraz z inną encją równoczesnie)
- zmiana w tescie metody TestOrderDaoFindAll() - było podwojne zapisanie do zmiennej order. Należło to rozdzielić na order i order 2
- dodanie zapisu koszyka do bazy dancyh cartDao.save(cart2)
- zmiana Long order2Id = order.get(id) na Long order2Id = order2.get(id)
(do dwóch zmiennych był przypisany taki sam Id z encji order a id order2 nie było do niczego przypisane)

2. CartDaoTestSuite:
- usunięcie podwójnego zapisu listy produktów w koszyku. Najpierw do koszyka był dodawany produkt a później do koszyk była przypisana pusta Arraylista
- usunięcie productDao.deleteById() encja produkt jest automatycznie usuwana z koszykiem

pszczepcio committed:
1. usunięcie Transactional z Dao
2. Zmiana w niektórych relacjach cascadeType. ALL na cascadeType.REFRESH
3. zmiana fetch.Eager na LAzy
4. Dodanie Transactional w Testach

pszczepcio committed:
- dodanie metody contractCartAndProductRelationshipTest() testującej  relację Cart i Product w testach CartDaoTestSuite
- dodanie metody  contractOrderAndProductRelationshipTest() testującej  relację Order i Product w testach OrderDaoTestSuite

pszczepcio committed:
- usuniecie zbędnych komentarzy w cartDaoTestSuite

KucharM committed:
Merge pull request #61 from klikuskodilla/naprawianieBledow

Naprawianie bledow

KucharM committed:
tymczasowe mappery - do implementacji

KucharM committed:
dao implementacja

KucharM committed:
serwisy z mapowaniem

KucharM committed:
serwisy i mappery

klikuskodilla reviewed (changes requested):
Implementacje związane z servisami jest ok. Dziwi mnie że zostały tutaj zaimplementowane po raz drugi klasy dla mapperów które oryginalnie są w tasku 39 który wisi od dłuższego czasu. Mappery nie są związane z taskiem 42 tylko z 39. Jeśli task 42 wymaga mapperów to trzeba dokończyć task 39 a nie pisać je na nowo.
![image](https://user-images.githubusercontent.com/33075397/58172775-c5b00b00-7c99-11e9-80f1-a947b328996c.png)

klikuskodilla commented:
Druga sprawa to dołączanie feature brancha, lepszym pomysłem jest poczekania aż branch naprawianieBłędów zostanie dołączony do mastera i wtedy domergować mastera. Dzięki czemu mamy pewność że jest on skończony
![image](https://user-images.githubusercontent.com/33075397/58173013-78806900-7c9a-11e9-8115-69fca29dc4d4.png)

pszczepcio commented:
Mi się wydaje, że ten task nie ma jednak sensu.  Po zaimplementowaniu userService dość sporo metod mi tam doszło wiec przynajmniej  UserService nie ma sensu wstępnie implementować.
